ocf_datapipes/select/drop_overnight_pv.py

```diff
@@ -1 +1,90 @@
 """Drop PV systems that report overnight"""
+import logging
+import random
+from datetime import datetime
+from typing import Dict, List, Union
+
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+logger = logging.getLogger(__name__)
+
+"""According to the UK Met office- https://www.metoffice.gov.uk/weather/learn-about/met-office-for-schools/other-content/other-resources/our-seasons
+
+Summer  - starts in June in the UK and finishes at the end of August.
+Autumn -  starts in September and finishes in November
+Winter - runs from December to February
+Spring - begins in March and ends in May
+According to worlddata.info - https://www.worlddata.info/europe/united-kingdom/sunset.php
+Month	    Sunrise	    Sunset      Hours of daylight
+January	    07:57 am	04:22 pm	8:25 h
+February	07:12 am	05:16 pm	10:05 h
+March	    06:12 am	06:06 pm	11:54 h
+April	    06:02 am	07:58 pm	13:56 h
+May	        05:06 am	08:47 pm	15:41 h
+June	    04:40 am	09:21 pm	16:41 h
+July	    04:59 am	09:13 pm	16:15 h
+August	    05:44 am	08:25 pm	14:41 h
+September	06:33 am	07:17 pm	12:44 h
+October	    07:22 am	06:09 pm	10:47 h
+November	07:16 am	04:13 pm	8:57 h
+December	07:57 am	03:53 pm	7:56 h
+"""
```

Comment:
good comments, I think you can use `[pvlib](https://pvlib-python.readthedocs.io/en/stable/)` to get this on this. 
If we want to do this, then we could use the location of each pv system to get the _night time_. 

However, this could be added as a comment, and we could update later